.github/workflows/deploy.yml

```diff
@@ -0,0 +1,48 @@
+name: Deploy
+on:
+  #push:
+  #  branches: [ main ]
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment"
+        required: true
+        default: "enable"
+        type: choice
+        options:
+        - nightly
+        - stable
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:  
+  deploy:
+    runs-on: ubuntu-latest
+    env:
+      DATA_ENV: ${{ github.event.inputs.environment || 'nightly' }}
+    permissions:
+      id-token: write
+      contents: read
+    
+    steps:      
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::490752553772:role/tlsn-deploy-slc
+          role-duration-seconds: 1800
+          aws-region: eu-central-1
+
+      - name: Cargo build
```

Comment:
Prior to this step, do we need to install the same Rust version (i.e. stable) as that in the ci.yml step here: https://github.com/tlsnotary/tlsn/blob/dev/.github/workflows/ci.yml#L60-L72 — to ensure the same Rust version is being used for building?

Also we used Swatinem/rust-cache@v2.5.0 in ci, will that help here to speeding up deployment?

Reply:
I also realised in ci.yml, we have used the same build command `cargo build --release` for notary server, if there is an easy way to utilise the built artefact there, then it ll be great — it's fine if it's too troublesome

Reply:
Done! (https://github.com/ntampakas/tlsn/commit/33641e1c614c43e74fd4854e4a21d32eda81ddf1)

Well, we could move the S3 copy step from the **cd-server.yml** workflow to the **ci.yml** one. This requires some modifications to the ci.yml workflow, such as using temporary security credentials and injecting the environment as well. If you are okay with it, I can start working on it.

Reply:
> Prior to this step, do we need to install the same Rust version (i.e. stable) as that in the ci.yml step here: https://github.com/tlsnotary/tlsn/blob/dev/.github/workflows/ci.yml#L60-L72 — to ensure the same Rust version is being used for building?
> 
> Also we used Swatinem/rust-cache@v2.5.0 in ci, will that help here to speed up deployment?

Done! (https://github.com/ntampakas/tlsn/commit/33641e1c614c43e74fd4854e4a21d32eda81ddf1)

Reply:
Hmm I think it's better to keep them separate to avoid complicating the ci workflow file, so good with building it again in GH runner :)